### PR TITLE
Fix bug in isMount and test more

### PR DESCRIPTION
**Pull request by lydia-duncan**
Previously, if a symlink was formed to a mountpoint, isMount would think that
symlink was also a mountpoint.  This is not correct behavior, with the rationale
being that if the original file was removed, the symlink would not maintain
the mount point.  Symlinks should have their link node compared instead.  My
method of finding the parent of a file also did not support this, as attaching
"/.." to the end of a symlink resulted in the parent of the location being
linked to instead of the parent of the location of the symlink itself.  Getting
around this required some string operations in the runtime which could be
replaced by functions in our Path module when they are implemented.  I
decided to fix the bug rather than not allow isMount to be called on links.

Added a test of this behavior, and a test to ensure that files and directories
which aren't mount points are recognized correctly.

The cleanfiles for the symlink mount test was having difficulty removing
the symlinks I created.  So I replaced it with a prediff that removes it.  This
is probably terrible and I am more than happy to back this change out,
pending reviewer disgruntlement.

### CODE REVIEW

File: runtime/src/chpl-file-utils.c [hunk 1]
```diff
@@ -164,16 +164,66 @@ qioerr chpl_fs_is_mount(int* ret, const char* name) {
...
+  char* safeNameCopy = (char* ) chpl_mem_allocMany(nameLen + 1, sizeof(char), CHPL_RT_MD_OS_LAYER_TMP_DATA, 0, 0);
+  strncpy(safeNameCopy, name, nameLen + 1);
+  // Need to copy name so that we can use it in the case of links
 
-  exitStatus = stat(name, &nBuf);
+  err = chpl_fs_is_link(&exitStatus, name);
+  if (err) {
+    // The stat call in is_link returned an error, which we would encounter too,
+    // so return immediately.
+    return err;
```
Comment (daviditen) [inline runtime/src/chpl-file-utils.c]: The strings parent and safeNameCopy don't get freed in these early returns for errors.

File: runtime/src/chpl-file-utils.c [hunk 2]
```diff
@@ -164,16 +164,66 @@ qioerr chpl_fs_is_mount(int* ret, const char* name) {
...
+  } else {
+    // We are not referring to a link, so concatenating "/.." is fine.
+    strncpy(parent, name, nameLen + 1);
+    strncat(parent, "/..", 3);
+    // TODO: Using "/" is not necessarily portable, look into this
+  }
+
+  exitStatus = lstat(name, &nBuf);
   if (exitStatus) {
     err = qio_mkerror_errno();
```
Comment (daviditen) [inline runtime/src/chpl-file-utils.c]: Here is another spot leaks parent and safeNameCopy.

File: test/modules/standard/FileSystem/lydia/isMount/symlinkToMount.prediff [hunk 3]
```diff
@@ -0,0 +1,12 @@
+#!/bin/bash
+
+# Note: I tried using a cleanfiles for this test, but it was failing to
```
Comment (daviditen) [inline test/modules/standard/FileSystem/lydia/isMount/symlinkToMount.prediff]: You could create and remove the symlinks internally to the chpl code (using FileSystem.symlink and FileSystem.remove) to get rid of the .preexec and .prediff files that create and delete them.  This is not required for this PR, but just an idea to clean this up.

File: runtime/src/chpl-file-utils.c (hunk 1 above)
Comment (lydia-duncan) [inline runtime/src/chpl-file-utils.c]: Whoops, thanks!

**Comment by daviditen** [comment]
Other than the two early returns that leak I think this is good.

**Comment by lydia-duncan** [comment]
I need to catch a bus, but I'll merge this when I get home.  That ought to give Travis enough time to start the silly check.

**Pull request merged**